fourkitchens/emulsify
# Adds function to replace large class functionality in each component

**Wouldn't it be lovely if instead of using different syntax for Drupal and Pattern Lab you could simply use this for adding classes `<div{{ bem('block') }}>`  and it just magically worked?**

And technically, the syntax is:

`<div{{ bem('block', (block_modifiers), block_blockname) }}>`

.. with the first being the blockname, the second being modifiers, and the third being a blockname override (modifiers and blockname optional). 💯 

**Changes:**
- Adds the bem function (`components/_twig-components/functions/bem.function.php`) to be able to pass in classes to Pattern Lab files (using simply `class=""`) and Drupal files (using `{{ attributes }}`) correctly.
- See `components/_patterns/01-atoms/02-text/00-headings/heading-1/heading-1.twig` and `components/_patterns/02-molecules/_block.twig` for an example usage of the new `bem` function. The block twig file is used directly in Drupal and correctly adds the bem class to the existing Drupal attributes

**Testing instructions:**
- Install and run vanilla D8 site
- Enable Emulsify as theme and checkout this branch (performing installation steps of course)
- Download https://github.com/evanmwillhite/unified-twig-extensions/tree/develop and enable Unified Twig Extensions module. 
- `npm start` to run Pattern Lab and verify [Header 1](http://localhost:3000/pattern-lab/public/?p=viewall-atoms-text) is accurately printing classes from file and yml
- Run Drupal site with preferred method and verify blocks are getting "block" class from `components/_patterns/02-molecules/_block.twig` as well as Drupal attributes.

**To-do:**
- Get forked changes into original [Unified Twig Extensions](https://github.com/drupal-pattern-lab/unified-twig-extensions/tree/develop) module
- Add bem function usage in all base files
- Update docs accordingly with second required Drupal module.
